Python 3.9 support [AttributeError: __qualname__]

When using MonkeyType with Python 3.9, I'm getting tons of error messages like
```
[...]
  File "/path/to/site-packages/monkeytype/encoding.py", line 82, in type_to_dict
    qualname = typ.__qualname__
  File "/usr/lib/python3.9/typing.py", line 640, in __getattr__
    raise AttributeError(attr)
AttributeError: __qualname__
```

So far for Arch Linux running the tests shows the following errors:
```
________________ TestTypeConversion.test_type_round_trip[typ7] _________________
tests/test_encoding.py:80: in test_type_round_trip
    assert type_from_dict(type_to_dict(typ)) == typ
monkeytype/encoding.py:130: in type_from_dict
    raise InvalidTypeError(
E   monkeytype.exceptions.InvalidTypeError: Attribute specified by 'Dict' in module 'typing' is of type <class 'typing._SpecialGenericAlias'>, not type.
________________ TestTypeConversion.test_type_round_trip[typ8] _________________
tests/test_encoding.py:80: in test_type_round_trip
    assert type_from_dict(type_to_dict(typ)) == typ
monkeytype/encoding.py:130: in type_from_dict
    raise InvalidTypeError(
E   monkeytype.exceptions.InvalidTypeError: Attribute specified by 'Dict' in module 'typing' is of type <class 'typing._SpecialGenericAlias'>, not type.
________________ TestTypeConversion.test_type_round_trip[typ9] _________________
```

--- a/monkeytype/compat.py
+++ b/monkeytype/compat.py
@@ -24,7 +24,7 @@ try:
 
     try:
         # Python 3.9
-        from typing import _SpecialGenericAlias
+        from typing import _SpecialGenericAlias  # type: ignore
 
         def is_generic(typ: Any) -> bool:
             return typ is Union or isinstance(typ, _GenericAlias) or isinstance(typ, _SpecialGenericAlias)

--- a/monkeytype/compat.py
+++ b/monkeytype/compat.py
@@ -22,8 +22,16 @@ try:
     def is_union(typ: Any) -> bool:
         return typ is Union or is_generic(typ) and typ.__origin__ is Union
 
-    def is_generic(typ: Any) -> bool:
-        return typ is Union or isinstance(typ, _GenericAlias)
+    try:
+        # Python 3.9
+        from typing import _SpecialGenericAlias
+
+        def is_generic(typ: Any) -> bool:
+            return typ is Union or isinstance(typ, _GenericAlias) or isinstance(typ, _SpecialGenericAlias)
+
+    except ImportError:
+        def is_generic(typ: Any) -> bool:
+            return typ is Union or isinstance(typ, _GenericAlias)
 
     def is_generic_of(typ: Any, gen: Any) -> bool:
         return is_generic(typ) and typ.__origin__ is gen.__origin__

